ArcGISRuntimeSDKQt_CppSamples/Analysis/DisplayUtilityAssociations/DisplayUtilityAssociations.cpp

```diff
@@ -0,0 +1,280 @@
+{
+  m_utilityNetwork = new UtilityNetwork(featureServerUrl, this);
+
+  // create symbols for the associations
+  m_attachmentSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Dot, Qt::green, 5, this);
+  m_connectivitySymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Dot, Qt::red, 5, this);
+
+  connectSignals();
+}
+
+void DisplayUtilityAssociations::addAssociations()
+{
+  // check if current viewpoint is outside the max scale
+  if(m_mapView->currentViewpoint(ViewpointType::CenterAndScale).targetScale() >= maxScale)
+  {
+    return;
+  }
+
+  // check if current viewpoint has a valid extent
+  Envelope extent = m_mapView->currentViewpoint(ViewpointType::BoundingGeometry).targetGeometry().extent();
+  if (!extent.isValid())
+  {
+    qDebug("Extent not valid");
+    return;
+  }
+
+  // get all the associations in the extent of the viewpoint
+  m_utilityNetwork->associations(extent);
+}
+
+DisplayUtilityAssociations::~DisplayUtilityAssociations() = default;
+
+void DisplayUtilityAssociations::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplayUtilityAssociations>("Esri.Samples", 1, 0, "DisplayUtilityAssociationsSample");
+}
+
+MapQuickView* DisplayUtilityAssociations::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DisplayUtilityAssociations::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_associationsOverlay);
+
+  // Get the image provider from the QML Engine
+  QQmlEngine* engine = QQmlEngine::contextForObject(this)->engine();
+  engine->addImageProvider(SymbolImageProvider::imageProviderId(), new SymbolImageProvider);
+  m_symbolImageProvider = static_cast<SymbolImageProvider*>(engine->imageProvider(SymbolImageProvider::imageProviderId()));
+
+  connect(m_mapView, &MapQuickView::setViewpointCompleted, this, [this](bool succeeded)
+  {
+    if (!succeeded)
+      return;
+
+    addAssociations();
+  });
+
+  connect(m_mapView, &MapQuickView::navigatingChanged, this, [this]()
+  {
+    if (!m_mapView->isNavigating())
+    {
+      addAssociations();
+    }
+  });
+
+  m_utilityNetwork->load();
+
+  emit mapViewChanged();
+}
+
+void DisplayUtilityAssociations::connectSignals()
+{
+  connect(m_utilityNetwork, &UtilityNetwork::doneLoading, this, [this](const Error& error)
+  {
+    if (!error.isEmpty())
+    {
+      qDebug() << error.message() << error.additionalMessage();
+      return;
+    }
+
+    m_mapView->setViewpoint(Viewpoint(Point(-9812698.37297436, 5131928.33743317, SpatialReference::webMercator()), targetScale));
+
+    // get all the edges and junctions in the network
+    QList<UtilityNetworkSource*> edges;
+    QList<UtilityNetworkSource*> junctions;
+
+    for (UtilityNetworkSource* networkSource : m_utilityNetwork->definition()->networkSources())
```

Comment:
I think you should get `m_utilityNetwork->definition()->networkSources()` up-front and cache it. There is a danger of becoming detached here I think